pkgs/stdenv/generic/setup.sh

```diff
@@ -3,6 +3,9 @@ set -o pipefail
 
 : ${outputs:=out}
 
+# If unset, assume the default hardening flags.
+: ${NIX_HARDENING_ENABLE="fortify stackprotector pic strictoverflow format relro bindnow"}
```

Comment:
Perhaps this should go in the setup-hook for cc-wrapper? `setup.sh` isn't exactly a paragon of clean layering, but I wouldn't want to make it worse.